cli/bpmetadata/cmd.go

```diff
@@ -203,3 +253,49 @@ func createContent(bpPath string, rootPath string, readmeContent []byte) Bluepri
 
 	return content
 }
+
+func writeMetadata(obj *BlueprintMetadata) error {
+	// marshal and write the file
+	yFile, err := yaml.Marshal(obj)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(metadataFileName, yFile, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func unmarshalMetadata(f []byte) (*BlueprintMetadata, error) {
+	var bpObj BlueprintMetadata
+	err := yaml.Unmarshal(f, &bpObj)
```

Comment:
Are comments preserved with yaml? I remember having to kyaml for comments.

Reply:
They may not be. I'll check

Reply:
b/2562073 to investigate and implement this.